### PR TITLE
#649 Add PO Contract When Activating Repo.

### LINKED ISSUE #649: Add PO Contract When Activating Repo
When a user activates their Repo, we should also add them as a Contributor and register a PO Contract (hourly rate $25 by default).

### PR DESCRIPTION
**Pull request by criske**
PR for #649 

closes #649

### CODE REVIEW

File: self-core-impl/src/test/java/com/selfxdsd/core/managers/StatusTestCase.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void repliesTaskNotRegisteredAndIssueClosed() {
                 + "@mihai this ticket is not registered as a task, "
                 + "therefore I'm not working on it at the moment.\n\n"
                 + "If you want me to take care of it, please reopen it."
-                + " However, reopening Issues is not encouraged --"
+                + " However, reopening Issues is not encouraged -- "
```
Comment (criske) [inline self-core-impl/src/test/java/com/selfxdsd/core/managers/StatusTestCase.java:134]: @amihaiemil small fix here -- this test was failing.

File: self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java [hunk 2]
```diff
@@ -148,6 +148,18 @@ public Project activate() {
...
+            this.owner.username(),
+            this.owner.provider().name()
+        );
+        project.contracts().addContract(
+            project.repoFullName(),
+            this.owner.username(),
+            this.owner.provider().name(),
+            BigDecimal.valueOf(2500),
+            Contract.Roles.PO
+        );
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java:162]: @criske I remember we once implemented a functionality, to automatically register the Contributor when adding a contract (if it's their first Contract in Self). Will not this work here? Maybe ``addContract(...)`` is enough. What do you think?

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@criske just one question

File: self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java (hunk 2 above)
Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java:162]: @amihaiemil Yep, you're right, just checked the storage impl :D

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**